### ISSUE TITLE
[Tcl] no syntax highlighting in TclOO blocks

### ISSUE THREAD

**Issue by anauthentic** [sublimehq/Packages#1055]
oo::define / oo::class -> methods looks like a plain text. 
![1](https://user-images.githubusercontent.com/10039408/27500901-1d883d86-5873-11e7-8863-ae203368426f.png)

**Comment by keith-hall** [comment]
please can you include the code from your screenshot in text format, for easier reproduction via copy and paste

**Comment by anauthentic** [comment]
ok, here is the code: https://pastebin.com/yS9apdN9

**Comment by RichardMullins** [comment]
I am having the same problem (although for me it only occurred after the most recent update), but I noticed some odd behaviour that could help to fix it.

If I add a brace into the wrong spot (like immediately after the oo::define) the rest of the formatting colour codes correctly.

so on line 42 above (and/or in the pastebin)
oo:define DB {

could be written as 
oo::define DB {}

which would fix the formatting (and break the code)

**Comment by RichardMullins** [comment]
OK I've discovered a few more things.

If you have a proc with something inside it (like a try) that opens new braces, the colour coding dies.

so

```
proc example {} {
  try {
    set a 10
  }
}
```

In this case the set will not be colour coded.

The try is not turned red, but blue. Red is the default on my setup for system commands and blue for commands that I have created (my own proc's etc). These seem to be the issue. oo::class et al are also in this second, blue, category.

I'm happy to help, but fairly time constrained. If I can work with someone, I can do the coding, just don't have time to get the syntax highlighting system in my head.
